pkg/http/http.go

```diff
@@ -5,32 +5,74 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 	"strings"
+
 	"wait4it/pkg/model"
 )
 
-func (h *HttpCheck) BuildContext(cx model.CheckContext) {
-	h.Url = cx.Host
-	h.Status = cx.HttpConf.StatusCode
+type checker struct {
+	url    string
+	status int
+	text   string
+}
+
+// NewChecker creates a new checker
+func NewChecker(c *model.CheckContext) (model.CheckInterface, error) {
+	checker := &checker{}
+	checker.buildContext(*c)
+	if err := checker.validate(); err != nil {
+		return nil, err
+	}
+
+	return checker, nil
+}
+
+func (c *checker) buildContext(cx model.CheckContext) {
+	c.url = cx.Host
+	c.status = cx.HttpConf.StatusCode
+
 	if len(cx.HttpConf.Text) > 0 {
-		h.Text = cx.HttpConf.Text
+		c.text = cx.HttpConf.Text
 	}
 }
 
-func (h *HttpCheck) Validate() error {
-	if !h.validateUrl() {
+func (c *checker) validate() error {
+	if !c.validateURL() {
 		return errors.New("invalid URL provided")
 	}
 
-	if !h.validateStatusCode() {
+	if !c.validateStatusCode() {
 		return errors.New("invalid status code provided")
 	}
 
 	return nil
 }
 
-func (h *HttpCheck) Check(ctx context.Context) (bool, bool, error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, h.Url, nil)
+func (c *checker) validateURL() bool {
+	_, err := url.ParseRequestURI(c.url)
+	if err != nil {
+		return false
+	}
+
+	u, err := url.Parse(c.url)
+	if err != nil || u.Scheme == "" || u.Host == "" {
```

Comment:
the schema should be `http` or `https` also both `Parse` and `ParseRequestURI` use the same function [parse](https://golang.org/src/net/url/url.go?#L508) the second parameter enfrce the schema, which you did in this line, so you can safely ignore the `ParseRequestURI`